public_html/script/src/drawable/drawable.js

```diff
@@ -34,12 +37,22 @@ export const Drawable = class {
                 'Drawable.addChild is being passed an object that doesn\'t inherit from Drawable',
                 'drawable.js');
         }
+        child.removeFromParent(this);
         child[PARENT] = this;
         this[CHILDREN].push(child);
         this.shouldRedraw = true;
         this.sortChildren();
     }
 
+    removeFromParent(newParent) { //call this BEFORE changing the parent
```

Comment:
`newParent` is there because subclasses can use it, and I left it the superclass implementation for clarity -- it doesn't do anything there but it makes it clear that `newParent` should be passed and that it might be used